torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +101,73 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
+        from .builder import wrap_fx_proxy
+        from .tensor import DynamicShapeVariable
+
+        assert not (
+            left.is_python_constant() and right.is_python_constant()
+        ), "Illegal generic list compare on constant lists"
+
+        # Most list-like variables implement comparison ops the same way,
+        # so they can re-use this helper.
+        # There are quirks though, like how `tuple([2]) == torch.Size([2])`,
+        # but `tuple([2]) != list([2])`
+        if len(left.items) != len(right.items):
+            return ConstantVariable(False, **options)
+        if len(left.items) == 0:
+            return ConstantVariable(True, **options)
+
+        # Generic list comparison works by iterating over left aka self and right the compared-to list.
+        # If we hit here, their lengths are the same and they cannot be expressed as python constants.
+        # So, we iterate over the zipped list items.
+        equal = True
+        list_type = None
+        comps = []
+        for l_r in zip(left.items, right.items):
+            l = l_r[0]
+            r = l_r[1]
+            # When iterating over the zipped items, we invoke compare.
+            # .compare must return a DynamicShapeVariable, a TensorVariable,  or raise unimplemented
+            # If unimplemented is raised, we rely on checkpointing state to roll us back and not write these operations to the graph
```

Comment:
Curiosity question: when does checkpointing happen? (Just thinking about it, I think my guess would be it's at the end of processing each python byte code, so that a single call to COMPARE_OP is "all-or-nothing" success (write every torch op to the graph) or failure (graph break).

Reply:
Grep for the checkpoint/Checkpointable structure :) 

`copy_graphstate` creates new ones